### PR TITLE
CORTX-34425: Add dockerfile for logstash

### PR DESCRIPTION
**Pull request by Sanket292001**
Signed-off-by: Sanket Salunke <sanket.s.salunke@seagate.com>

# Problem Statement
- Add a dockerfile to create a custom image from the base logstash image.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [ ] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] Jira ID/ COMMUNITY is prefixed in PR title
      Prefix Syntax - 
        `CORTX-<5 digit JIRA ID>: <PR Title>` - for Seagate Employees
        `COMMUNITY: <PR Title>`  - for open source contributors
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by Sanket292001** [comment]
I have resolved Codacy issues.
It is showing Codacy status failed and +1 in issues. But, not showing any issue under the issues section.
It is displaying "You don't have new issues."

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
@Sanket292001 Please move this to `solutions/dashboard/build` directory at `solutions/dashboard/build/logstash/Dockerfile`  
Update `docker-compose.yml` for logstash and update `build.sh` to generate the `logstash` image.

**Comment by Sanket292001** [comment]
Hi @shailesh-vaidya ,
Modified the directory structure and other files. Also, updated the old Logstash image name with the new one in the YAML files.
Please check

**Review by shailesh-vaidya — APPROVED** [review]

File: solutions/dashboard/build/docker-compose.yml [hunk 1]
```diff
@@ -40,3 +40,8 @@ services:
     build:
       context: ../tools/jenkins/
       dockerfile: ./Dockerfile
+  logstash:
+    image: logstash:$TAG
+    build:
+      context: ./logstash/
```
Comment (gauravchaudhari02) [inline solutions/dashboard/build/docker-compose.yml:46]: @Sanket292001 , why are we not putting logstash related files in `tools` folder like other tools?

Comment (gauravchaudhari02) [reply solutions/dashboard/build/docker-compose.yml:46]: @Sanket292001, as discussed, lets do tools folder management in next PR

**Review by gauravchaudhari02 — APPROVED** [review]

**Comment by Sanket292001** [comment]
Closed this PR as new one is raised and merged

**Pull request closed without merging**